wireshark_dissector/ambit-dissector.c

```diff
@@ -3194,16 +3264,24 @@ proto_register_ambit(void)
 
     proto_register_field_array(proto_ambit, hf, array_length(hf));
     proto_register_subtree_array(ett, array_length(ett));
-    //register_dissector("ambit", dissect_ambit, proto_ambit);
+    ambit_handle = register_dissector("ambit", dissect_ambit, proto_ambit);
+
+    // Function not called by the API, why ???
+    proto_reg_handoff_ambit();
 }
 
 void
 proto_reg_handoff_ambit(void)
 {
-    static dissector_handle_t ambit_handle;
-
-    //ambit_handle = find_dissector("ambit");
-    ambit_handle = new_create_dissector_handle(dissect_ambit, proto_ambit);
+    /* ambit_handle = find_dissector("ambit"); */
+    /* ambit_handle = create_dissector_handle(dissect_ambit, proto_ambit); */
     dissector_add_uint("usb.interrupt", IF_CLASS_UNKNOWN, ambit_handle);
     dissector_add_uint("usb.interrupt", IF_CLASS_HID, ambit_handle);
 }
+
+#define PLUGIN_VERSION "1.0.0"
+
+WS_DLL_PUBLIC_DEF const gchar plugin_version[] = PLUGIN_VERSION;
+WS_DLL_PUBLIC_DEF const int plugin_want_major = 3;
+WS_DLL_PUBLIC_DEF const int plugin_want_minor = 4;
```

Comment:
Ubuntu LTS still has version 3.2.3 of wireshark, should we lowering the minimum version to that? It seems to compile fine with 3.2.3.

Note: I have no experience with Wireshark so don't know much about how the version here is applied/checked/enforced in Wireshark.

Reply:
I'm running Debian testing with Wireshark 3.4. It should be ok for wireshark 3.2 as we use basic API which doesn't change (deprecated functions were removed).
I just made another commit to support Wireshark version 3.2, but I didn't tested it, if someone could do it...
The bad thing here is that you have to set a specific target version, not a minimal one.